docs: add migration guide from v6 to v7 (#1246)
faker-js/faker: docs/guide/migration-guide-v6.md

# Migrating from Faker v6 to v7

[[toc]]

### Node 12 no longer supported

> You need at least Node 14 to use faker.

Reviewer: ```suggestion
You need at least Node 14 to use Faker.
```

> You require at least Node 14 to run faker

Reviewer 2: ```suggestion
You need at least Node 14 to use faker.
```

### Default export removed

If you were previously importing faker like this:

```js
import faker from '@faker-js/faker';
```

> You should instead use:

Reviewer: ```suggestion
You must instead use:
```

> You should instead use

Reviewer 2: ```suggestion
You should instead use:
```

```js
import { faker } from '@faker-js/faker';
```

### Deprecated methods changed

> Several methods were renamed or moved.

Reviewer 2: Not sure about this one:

```suggestion
Several methods were renamed or moved:
```
